password-hash/src/pepper.rs

```diff
@@ -0,0 +1,16 @@
+use crate::{Ident, Output};
+
+/// Trait for hash peppering functions.
+pub trait Pepper {
+    /// Algorithm-specific errors
+    type Error;
+
+    /// Peppers the `data`.
+    fn pepper(&self, data: Output) -> Result<Output, Self::Error>;
+
+    /// Verifies that the `data` is peppered to the same as `to_verify`.
+    fn verify(&self, data: Output, to_verify: &Output) -> Result<(), Self::Error>;
+
+    /// Identifies the algorithm uniquely.
+    fn ident<'a>(&'a self) -> Ident<'a>;
```

Comment:
I would like to make it `fn ident() -> Ident<'static>` but eg. the `str "HMAC-SHA256"` can only produced by something hacky like [const-concat](https://lib.rs/crates/const-concat) or the a bit more professional [const_format](https://lib.rs/crates/const_format).